weTranslate/SearchHeaderView.swift

```diff
@@ -0,0 +1,124 @@
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setTitle("French", forState: .Normal)
+        button.setTitleColor(UIColor.whiteColor(), forState: .Normal)
+        button.addTarget(self, action: "selectToLanguage:", forControlEvents: .TouchUpInside)
+        return button
+    }()
+
+    private lazy var swapButton: UIButton = {
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setImage(UIImage(named: "Swap"), forState: .Normal)
+        button.addTarget(self, action: "swap:", forControlEvents: .TouchUpInside)
+        return button
+    }()
+
+    private let languagesPickerView: UIPickerView = {
+        let pickerView = UIPickerView()
+        pickerView.translatesAutoresizingMaskIntoConstraints = false
+        pickerView.hidden = true
+        return pickerView
+    }()
+
+    private let languageView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .Horizontal
+        stackView.spacing = 10
+        return stackView
+    }()
+
+    private let bodyView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .Vertical
+        stackView.spacing = 25
+        stackView.layoutMargins = UIEdgeInsets(top: 25, left: 25, bottom: 25, right: 25)
+        stackView.layoutMarginsRelativeArrangement = true
+        return stackView
+    }()
+
+
+    // MARK: - Initializers
+
+    override init(frame: CGRect = .zero) {
+        super.init(frame: frame)
+
+        languageView.addArrangedSubview(fromLanguageButton)
+        languageView.addArrangedSubview(swapButton)
+        languageView.addArrangedSubview(toLanguageButton)
+
+        bodyView.addArrangedSubview(searchTextField)
+        bodyView.addArrangedSubview(languageView)
+        bodyView.addArrangedSubview(languagesPickerView)
+        addSubview(bodyView)
+
+        let margins = layoutMarginsGuide
+        bodyView.leadingAnchor.constraintEqualToAnchor(margins.leadingAnchor).active = true
+        bodyView.trailingAnchor.constraintEqualToAnchor(margins.trailingAnchor).active = true
+        bodyView.topAnchor.constraintEqualToAnchor(margins.topAnchor).active = true
+        bodyView.bottomAnchor.constraintEqualToAnchor(margins.bottomAnchor).active = true
+
+        searchTextField.heightAnchor.constraintEqualToConstant(44).active = true
+        fromLanguageButton.heightAnchor.constraintEqualToConstant(25).active = true
+        fromLanguageButton.widthAnchor.constraintEqualToAnchor(toLanguageButton.widthAnchor, multiplier: 1.0).active = true
+    }
+
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+
+    // MARK: - Methods
+
+    func selectFromLanguage(sender: UIButton?) {
+        UIView.animateWithDuration(0.3) {
```

Comment:
What about creating a constant for 0.3 that could be use in the animation display/hide